bezier/src/cubic_bezier.rs

```diff
@@ -183,6 +183,140 @@ impl CubicBezierSegment {
         find_cubic_bezier_inflection_points(self)
     }
 
+    /// Return local x extremum points or None if this curve is monotone.
+///
+/// This returns the advancements along the curve, not the actual x position.
+    pub fn find_x_extremums(&self) -> UpToTwo<f32> {
+        let mut ret = UpToTwo::new();
+        // See www.faculty.idc.ac.il/arik/quality/appendixa.html for an explanation
+        // The derivative of a cubic bezier curve is a curve representing a second degree polynomial function
+        // f(x) = a * x² + b * x + c such as :
+        let a = 3.0 * (self.to.x - 3.0 * self.ctrl2.x + 3.0 * self.ctrl1.x - self.from.x);
+        let b = 6.0 * (self.ctrl2.x - 2.0 * self.ctrl1.x + self.from.x);
+        let c = 3.0 * (self.ctrl1.x - self.from.x);
+
+        println!("a : {} | b : {} | c : {}", a, b, c);
+
+        // If the derivative is a linear function
+        if a == 0.0 {
+            if b == 0.0 {
+                // If the derivative is a constant function
+                if c == 0.0 {
+                    ret.push(0.0);
+                }
+            } else {
+                ret.push(-c / b);
+            }
+            return ret;
+        }
+
+        fn in_range(t: f32) -> bool { t >= 0.0 && t < 1.0 }
+
+        let discriminant = b * b - 4.0 * a * c;
+
+        println!("delta : {} ", discriminant);
+
+
+        // There is no Real solution for the equation
+        if discriminant < 0.0 {
+            return ret;
+        }
+
+        // There is one Real solution for the equation
+        if discriminant == 0.0 {
+            let t = -b / (2.0 * a);
+            if in_range(t) {
+                ret.push(t);
+            }
+            return ret;
+        }
+
+        // There are two Real solutions for the equation
+        let discriminant_sqrt = discriminant.sqrt();
+        println!("delta_sqrt : {} ", discriminant_sqrt);
+
+        let first_extremum = (-b - discriminant_sqrt) / (2.0 * a);
+        let second_extremum = (-b + discriminant_sqrt) / (2.0 * a);
+
+        println!("first_extremum : {} , second_extremum : {}", first_extremum, second_extremum);
+
+        if in_range(first_extremum) {
+            ret.push(first_extremum);
+        }
+
+        if in_range(second_extremum) {
+            ret.push(second_extremum);
+        }
+        ret
+    }
+
+    /// Return local y extremum points or None if this curve is monotone.
+    ///
+    /// This returns the advancements along the curve, not the actual y position.
+    pub fn find_y_extremums(&self) -> UpToTwo<f32> {
```

Comment:
In order to avoid duplicating this code, you could transpose the curve (flip x and y of all its points), and call find_x_extremums, [like solve_t_for_y does](https://github.com/nical/lyon/blob/9bc8ada976994d213667abf0d7a3f006b1f188b5/bezier/src/cubic_bezier.rs#L338).